core/src/loader.rs

```diff
@@ -38,6 +40,50 @@ use url::form_urlencoded;
 
 pub type Handle = Index;
 
+/// How Ruffle should load movies.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, Default)]
+pub enum LoadBehavior {
+    /// Allow movies to execute before they have finished loading.
+    ///
+    /// Frames/bytes loaded values will tick up normally and progress events
+    /// will be fired at regular intervals. Movie preload animations will play
+    /// normally.
+    #[serde(rename = "streaming")]
+    #[default]
+    Streaming,
+
+    /// Delay execution of loaded movies until they have finished loading.
+    ///
+    /// Movies will see themselves load immediately. Preload animations will be
+    /// skipped. This may break movies that depend on loading during execution.
+    #[serde(rename = "delayed")]
+    Delayed,
+
+    /// Block Ruffle until movies have finished loading.
+    ///
+    /// This has the same implications as `Delay`, but tag processing will be
+    /// done synchronously. Complex movies will visibly block the player from
+    /// accepting user input and the application will appear to freeze.
+    #[serde(rename = "blocking")]
+    Blocking,
+}
+
+impl FromStr for LoadBehavior {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
I'm kinda surprised serde doesn't generate this for us

Reply:
Unfortunately nope! I even double-checked by removing the `FromStr` impl.